shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/watchers/DeploymentEventSource.java

```diff
@@ -0,0 +1,83 @@
+package com.redhat.service.bridge.shard.operator.watchers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.bridge.shard.operator.utils.LabelsBuilder;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.WatcherException;
+import io.javaoperatorsdk.operator.processing.event.AbstractEventSource;
+
+import static io.javaoperatorsdk.operator.processing.KubernetesResourceUtils.getUID;
+import static io.javaoperatorsdk.operator.processing.KubernetesResourceUtils.getVersion;
+
+/**
+ * Used by the controllers to watch changes on Deployment objects.
+ *
+ */
+public class DeploymentEventSource extends AbstractEventSource implements Watcher<Deployment> {
+    private static final Logger log = LoggerFactory.getLogger(DeploymentEventSource.class);
+
+    private final KubernetesClient client;
+
+    private final String applicationType;
+
+    public static DeploymentEventSource createAndRegisterWatch(KubernetesClient client, String applicationType) {
+        DeploymentEventSource deploymentEventSource = new DeploymentEventSource(client, applicationType);
+        deploymentEventSource.registerWatch(applicationType);
+        return deploymentEventSource;
+    }
+
+    private DeploymentEventSource(KubernetesClient client, String applicationType) {
+        this.client = client;
+        this.applicationType = applicationType;
+    }
+
+    private void registerWatch(String applicationType) {
+        client
+                .apps()
+                .deployments()
+                .inAnyNamespace()
+                .withLabel(LabelsBuilder.MANAGED_BY_LABEL, LabelsBuilder.OPERATOR_NAME)
+                .withLabel(LabelsBuilder.APPLICATION_TYPE_LABEL, applicationType)
```

Comment:
Related to the previous comment: the controller specifies the `applicationType` of the Deployment it would like to monitor. So the `BridgeIngressController` will use `LabelsBuilder.BRIDGE_INGRESS_APPLICATION_TYPE`, and the `ProcessorController` (to be implemented) will use something like `LabelsBuilder.EXECUTOR_APPLICATION_TYPE`.

Reply:
That's fine. But consider using the `LabelsBuilder` fluent API instead. These constants should be private.

Reply:
I've moved the `LabelsBulder$build` method under `LabelsBuilder$buildWithDefaults`. The new `build` method will just return the labels without any addition: in this way we can use the fluent api to build labels that do not include the default ones.

Reply:
Moved to LabelsBuilder